src/azure-cli/azure/cli/command_modules/reservations/custom.py

```diff
@@ -35,3 +38,30 @@ def cli_reservation_merge_reservation(client, reservation_order_id,
     return client.merge(reservation_order_id,
                         [create_resource_id(reservation_order_id, reservation_id_1),
                          create_resource_id(reservation_order_id, reservation_id_2)])
+
+
+def cli_calculate(client, sku, location, reserved_resource_type, billing_scope_id, term,
+                  billing_plan, quantity, applied_scope_type, display_name, applied_scopes=None,
+                  renew=False, reserved_resource_properties=None):
+    sku_name = SkuName(name=sku)
+    properties = PurchaseRequestPropertiesReservedResourceProperties(instance_flexibility=reserved_resource_properties)
+    body = PurchaseRequest(sku=sku_name, location=location, reserved_resource_type=reserved_resource_type,
+                           billing_scope_id=billing_scope_id, term=term, quantity=quantity,
+                           display_name=display_name,
+                           applied_scope_type=applied_scope_type,
+                           applied_scopes=applied_scopes, billing_plan=billing_plan,
+                           renew=renew, reserved_resource_properties=properties)
+    return client.calculate(body)
+
+
+def cli_purchase(client, reservation_order_id, sku, location, reserved_resource_type, billing_scope_id, term,
+                 billing_plan, quantity, applied_scope_type, display_name, applied_scopes=None,
```

Comment:
Your billing_plan is not displayed as required parameter. What if no --billing-plan provided? 
![image](https://user-images.githubusercontent.com/49508232/69249995-d1996d00-0be9-11ea-8bcb-3df97087180f.png)

Reply:
Default as "Upfront", Not all resource type supports the billing plan. This is an optional value, like renew